src/app/icd/client/ICDClientStorage.h

```diff
@@ -61,16 +61,31 @@ class ICDClientStorage
      * Store updated ICD ClientInfo to storage when ICD registration completes or check-in message
      * comes.
      *
-     * @param[in] aICDClientInfo the updated ICD Client Info.
+     * @param[in] clientInfo the updated ICD Client Info.
      */
     virtual CHIP_ERROR StoreEntry(const ICDClientInfo & clientInfo) = 0;
 
+    /**
+     * Remove ICD key from clientInfo when ICD registration fails
+     *
+     * @param[inout] clientInfo the updated ICD Client Info.
+     */
+    virtual void RemoveKey(ICDClientInfo & clientInfo) = 0;
+
+    /**
+     * Get ICD ClientInfo from storage
+     * One user case is to retrieve UserActiveModeTriggerHint and inform how user how to wake up sleepy device.
```

Comment:
s/user/use/

And

s/inform how user how to wake up sleepy/inform the user how to wake up a sleepy/ (note: multiple grammar fixes here).